### PR TITLE
[ioctl] Cleanup TestNewNodeDelegateCmd

### PR DESCRIPTION
**Pull request by LuckyPigeon**
# Description

Cleanup `TestNewNodeDelegateCmd` with following adjustment.

* Remove redundant mock test
* Replace `AnyTimes` with specific `Times`
* Move specific `client` and variable to correspond `t.Run`

## Type of change
Please delete options that are not relevant.
- [x] Code refactor or improvement

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [x] TestNewNodeDelegateCmd

**Test Configuration**:
- Firmware version: Ubuntu 21.04
- Hardware:
- Toolchain:
- SDK:

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: ioctl/newcmd/node/nodedelegate_test.go [hunk 1]
```diff
@@ -1,13 +1,19 @@
...
 
 import (
 	"testing"
 
 	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/require"
+
 	"github.com/iotexproject/iotex-proto/golang/iotexapi"
 	"github.com/iotexproject/iotex-proto/golang/iotexapi/mock_iotexapi"
 	"github.com/iotexproject/iotex-proto/golang/iotextypes"
```
Comment (huof6829) [inline ioctl/newcmd/node/nodedelegate_test.go:16]: these packages should be in one group

Comment (huof6829) [inline ioctl/newcmd/node/nodedelegate_test.go:16]: these packages should be in one group

**Comment by huof6829** [comment]
nodedelegate.go#L99 `var err error`  can be delete

**Comment by LuckyPigeon** [comment]
`var err error` in `nodedelegate.go` #L102 is used in
https://github.com/iotexproject/iotex-core/blob/623be4117c0694d464a17bfd54518cec29b8b4f8/ioctl/newcmd/node/nodedelegate.go#L240-L243

**Comment by huof6829** [comment]
> I have added `epochStartHeight` at #3411 once, but @Liuhaai tells me to revert, and I'm not sure we need to add it or not. If it's a necessary, I'll add it both here and #3411.

I'v modified #3411.  After #3411 merged, you can use getProbationList().

**Comment by huof6829** [comment]
> I have added `epochStartHeight` at #3411 once, but @Liuhaai tells me to revert, and I'm not sure we need to add it or not. If it's a necessary, I'll add it both here and #3411.

I'v modified #3411.  After #3411 merged, you can use getProbationList().

**Comment by huof6829** [comment]
> `var err error` in `nodedelegate.go` #L102 had been used at
> 
> https://github.com/iotexproject/iotex-core/blob/623be4117c0694d464a17bfd54518cec29b8b4f8/ioctl/newcmd/node/nodedelegate.go#L240-L243

fixed

**Review by huof6829 — APPROVED** [review]

File: ioctl/newcmd/node/nodedelegate.go [hunk 2]
```diff
@@ -85,6 +85,9 @@ func NewNodeDelegateCmd(client ioctl.Client) *cobra.Command {
 		insecure  bool
 	)
 
+	_nodeStatus = map[bool]string{true: "active", false: "false"}
```
Comment (Liuhaai) [inline ioctl/newcmd/node/nodedelegate.go:88]: could you use a subfunc to replace this map design?

File: ioctl/newcmd/node/nodedelegate.go [hunk 3]
```diff
@@ -260,7 +249,7 @@ func (m *nextDelegatesMessage) String(epochNum uint64) string {
 	lines = append(lines, fmt.Sprintf(formatTitleString, "Address", "Rank", "Alias", "Status", "Votes"))
 	for _, bp := range m.Delegates {
 		lines = append(lines, fmt.Sprintf(formatDataString, bp.Address, bp.Rank,
```
Comment (Liuhaai) [inline ioctl/newcmd/node/nodedelegate.go:251]: if bp.Active {}

File: ioctl/newcmd/node/nodedelegate.go [hunk 4]
```diff
@@ -280,7 +269,24 @@ func (m *delegatesMessage) String() string {
 		"Address", "Rank", "Alias", "Status", "Blocks", "ProbatedStatus", "Votes"))
 	for _, bp := range m.Delegates {
 		lines = append(lines, fmt.Sprintf(formatDataString, bp.Address, bp.Rank,
```
Comment (Liuhaai) [inline ioctl/newcmd/node/nodedelegate.go:271]: same

File: ioctl/newcmd/node/nodedelegate.go [hunk 5]
```diff
@@ -280,7 +269,24 @@ func (m *delegatesMessage) String() string {
...
-			bp.Alias, _nodeStatus[bp.Active], bp.Production, _probatedStatus[bp.ProbatedStatus], bp.Votes))
+			bp.Alias, ConfigMap("node", bp.Active), bp.Production, ConfigMap("probated", bp.ProbatedStatus), bp.Votes))
 	}
 	return strings.Join(lines, "\n")
 }
+
+// ConfigMap return the map of status configuration
+func ConfigMap(t string, b bool) string {
```
Comment (Liuhaai) [inline ioctl/newcmd/node/nodedelegate.go:278]: overkilled

**Review by CoderZhi — APPROVED** [review]

**Pull request merged**